gtsfm/utils/tracks.py

```diff
@@ -0,0 +1,72 @@
+"""Utilities for 2D and 3D tracks.
+
+Authors: Ayush Baid
+"""
+from typing import Dict, List
+
+from gtsam import Cal3Bundler, SfmTrack
+
+from gtsfm.common.sfm_track import SfmMeasurement, SfmTrack2d
+from gtsfm.data_association.point3d_initializer import (
+    Point3dInitializer,
+    TriangulationParam,
+    TriangulationExitCode,
+)
+
+
+def classify_tracks2d_with_gt_cameras(
+    tracks: List[SfmTrack2d], cameras_gt: List[Cal3Bundler], reproj_error_thresh_px: float = 3
+) -> List[TriangulationExitCode]:
+    """Classifies the 2D tracks w.r.t ground truth cameras by performing triangulation and collecting exit codes.
+
+    Args:
+        tracks: list of 2d tracks, of length J.
+        cameras_gt: cameras with GT params.
+        reproj_error_thresh_px (optional): Reprojection error threshold (in pixels) for a track to be considered an
+                                           all-inlier one. Defaults to 3.
+
+    Returns:
+        The triangulation exit code for each input track, as list of length J (same as input).
```

Comment:
Not sure if J is necessary if youre clarifying this in brackets anyway. It would be useful if you had some expression in terms of J, but looks like not.

Reply:
removed it. Thanks